### DISCUSSION TITLE
How to re-authenticate?

### DISCUSSION THREAD

**Discussion by bradlindblad** [community/community#6711] [Copilot Conversations] [answered]
I authenticated the app with the wrong github account, and after restarting and reinstalling the extension, I cannot get the prompt to authenticate with my correct github account.

Is there a way to re-authenticate?

**Comment by avigloz** [comment]
Having the same issue; asked me for a PAT once, failed (my fault), then never asked again despite multiple uninstall-reinstalls (of the extension). This is all I get now: 
![image](https://user-images.githubusercontent.com/5449202/138729862-d7c71f37-1edb-47a5-946e-2569703006e4.png)

(Sidenote: I am using VSCodium on Linux)

**Reply by SOV37** [reply]
Root

**Comment by olliethedev** [comment]
In your Keychain if you are on a Mac, find `vscodevscode.github-authentication` and delete it. Restart you IDE and it will prompt to login for copilot extension.

**Comment by drifkin** [comment]
If you haven't already, try logging out of the other Github account by clicking on the silhouette icon and then restarting VS Code.
![image](https://user-images.githubusercontent.com/175530/138767123-9788b2de-a273-4829-ba48-878969e40f4f.png)

**Reply by bradlindblad** [reply]
This worked to authenticate for me, but when I try to manually turn on copilot, I keep getting an error: 
![image](https://user-images.githubusercontent.com/34219856/138770221-8096117c-5861-4159-9c0d-995d6eafda8c.png)

**Reply by juanmespina** [reply]
I did this in windows10 and it worked! Thanks!

**Reply by vncafecode** [reply]
I'm windows 10 but not logged in to any account. How can I re-authenticate? 

![image](https://user-images.githubusercontent.com/44223474/139617692-01002406-094a-4853-9a9d-a705266dbfc3.png)

**Comment by davecheney** [comment]
@bradlindblad can you paste the output from the GitHub copilot output window. Thank you.

**Reply by bradlindblad** [reply]
Where is the github copilot output window?

**Reply by davecheney** [reply]
View -> Output, select `Github Copilot` on the right.

**Reply by bradlindblad** [reply]
![image](https://user-images.githubusercontent.com/34219856/138879385-a1a790e0-4e87-4349-a567-556a9c0ef375.png)

**Reply by avigloz** [reply]
In my case, this is the output:
![image](https://user-images.githubusercontent.com/5449202/138923466-6c06102e-d6ac-4178-a7fb-e2ffb042aacb.png)

If it's missing a token, I feel like it should ask me to provide a new one, no? In any case, seems like authentication is generally broken.

This occurs even after re-installing the extension/signing out of Github via Accounts and re-entering a PAT.

**Reply by rolimans** [reply]
I'm having the same issue, are you using VSCode or VSCodium?

**Reply by bradlindblad** [reply]
> I'm having the same issue, are you using VSCode or VSCodium?

VSCode on Windows10

**Reply by Kayoscape** [reply]
I am also having this issue on VSCode for Windows 10. I installed the extension and linked my GitHub account before I was accepted into the test program.

**Reply by jamaldevsnet** [reply]
Same here on ubuntu vscode!

**Reply by avigloz** [reply]
@davecheney any update on this issue?

**Reply by Kayoscape** [reply]
I recently tried signing out of GitHub and re-authenticating, but it did not fix this issue. Has anyone found a solution?

**Reply by avigloz** [reply]
@Kayoscape nope, and considering how long it's been, the devs seem to have decided to ignore this issue

**Reply by hsenag** [reply]
@avigloz @Kayoscape @jamaldevsnet Sorry for the gap in our responses, we're quite a small team at present.

We have actually been working on making it easier to debug when something goes wrong with sign in. In the latest version, released on Friday, we added a status indicator for this: <img width="377" alt="Screenshot 2021-11-18 at 12 55 16" src="https://user-images.githubusercontent.com/5427083/142416943-7aea2041-67fd-4606-ba9f-115b82a3652e.png">.

Do you see the little "crossed-out" Copilot icon? If so could you try clicking on it and reporting what it says?

**Reply by Kayoscape** [reply]
@hsenag Here is the message: "GitHub Copilot could not connect to server. Extension activation failed: "User not authorized""

**Reply by hsenag** [reply]
@Kayoscape Thanks. I'm afraid I don't know what's wrong right now, but I hope to improve the diagnostics for this case a bit more soon. I'll ping you when that's done and you can try again.

**Reply by Kayoscape** [reply]
It's been another two months and I haven't heard anything about a potential fix for this issue. Nothing has changed in VSCode, and I am still unable to use GitHub Copilot despite being in the program. @hsenag, where is that ping you promised?

**Reply by avigloz** [reply]
In the same boat as @Kayoscape. Please let us know what the status is, @hsenag.

**Reply by Kayoscape** [reply]
With @davecheney's answer, I was able to finally get it working. Here's how I did it:

1. Sign out of VSCode (using the method listed in @davecheney's answer).
2. Uninstall the GitHub Copilot extension.
3. Restart VSCode.
4. Install the GitHub Copilot extension.
5. Sign in from the Copilot icon at the bottom of the window and allow the browser link to open in VSCode after authenticating.

Thank you to everyone who worked on this.

**Reply by davecheney** [reply]
Uninstalling the extension is unnecessary.

**Comment by davecheney** [comment, marked as answer]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by mark-wiemer** [reply]
I'm not seeing that icon anymore, have the steps changed in the past year @davecheney? I'm having the same problem as the original report

**Reply by mark-wiemer** [reply]
Nevermind, it turns out I had hidden the "Accounts" icon from my activity bar. Once I right-clicked on the activity bar I saw the option to show the "Accounts" icon, reloaded VS Code, and everything worked.

![image](https://user-images.githubusercontent.com/7833360/218599396-ec1f4368-5d70-4ced-b7df-3bc6d8b4c5db.png)

**Comment by maheshj01** [comment]
My Github copilot is always loading, what could be the reason it was working fine a  few days ago. I did re-authenticate too but it still keeps spinning

<img width="263" alt="image" src="https://user-images.githubusercontent.com/31410839/152568331-7b315b19-0a94-4fa8-a80b-f9c2c0ffadb7.png">

**Reply by davecheney** [reply]
This sounds like a different problem. Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by maheshj01** [reply]
sorry for commenting here,
Discussion continued on the relevant thread https://github.com/github/feedback/discussions/11022
